plus/impl/src/main/java/org/sakaiproject/plus/impl/PlusServiceImpl.java

```diff
@@ -725,6 +746,20 @@ public void syncSiteMemberships(String contextGuid, Site site) throws LTIExcepti
 			cLog.setStatus("Exception processing Names and Roles data="+e.getMessage());
 		}
 
+		// Clean up inactive entries (i.e. like a week since the last NRPS retrieval)
+		// We may be in a thread / task instead of a login context...
+		pushAdvisor();
+		try {
+			int minutes = getInactiveExpireMinutes(context);
+			List<Membership> deleted_memberships = removeSiteUsersMinutesOld(context, minutes);
+			if ( deleted_memberships.size() > 0 ) {
+				log.info("Inactive memberships removed {} from {}", deleted_memberships.size(), site.getId());
+				dbs.append("Inactive memberships removed="+deleted_memberships.size());
+			}
+		} finally {
+			popAdvisor();
```

Comment:
this is a real old way to do advisors

preferred way is to create advisor, pass it to pushAdvisor and then pass it back to popAdvisor

Reply:
I will fix this  Thanks.